next.config.js

```diff
@@ -90,12 +91,14 @@ module.exports = withSwingset({
 		},
 		async redirects() {
 			const { simpleRedirects, globRedirects } = await redirectsConfig()
+			const tutorialRedirects = await getTutorialRedirects()
+
 			await fs.promises.writeFile(
 				path.join('src', 'data', '_redirects.generated.json'),
 				JSON.stringify(simpleRedirects, null, 2),
 				'utf-8'
 			)
-			return globRedirects
+			return [...globRedirects, ...tutorialRedirects]
```

Comment:
can we shift the tutorial redirects fetching into `redirectsConfig()` in `build-libs/redirects.js`? The idea being we keep the redirect generation logic consolidated and consistent.

Reply:
I'd like this file to stay unchanged. The redirects would get split yea, but it should all be handled correctly by the existing setup!

Reply:
ah oops, I missed that we have `splitRedirectsByType` to do the work for us 👍 👍 👍 I think I'm tracking now, like we'd add `const tutorialRedirects = await getTutorialRedirects()` in right after [proxiedSiteRedirects](https://github.com/hashicorp/dev-portal/blob/a15f17e311880b0362ccaee170bea5c7bd79b399/build-libs/redirects.js#L442), right?

Reply:
Yeah, exactly! 😊

Reply:
ah yes, I can do that!

Reply:
@BRKalow since the `simple redirects` are written to the generated file to be served from middleware and have docs-specific logic handled there, I'm thinking of creating a `tutorialRedirects` return value similar to what @zchsh suggested yesterday. So I can just spread here alongside the glob.

Or is there something I'm missing about wanting to serve the simple redirects in middleware?

Reply:
It's mostly just about being consistent with how we handle redirects. If there's no functional reason to have special handling for the tutorials redirects, it seems reasonable to handle them like all other redirects. The current logic should handle the tutorial redirects without issue. We initially did the simple/glob split to try and limit the number of redirects defined in our config.

Reply:
The issue is that the current setup is very 'product' driven and learn urls don't necessarily abide by that so some refactoring would need to be done I think to make it work. I could look into it.

Reply:
Unless maybe the learn redirects just go under '*', so maybe im mistaken!

Reply:
@kendallstrautman I think it will work as-is. 💭  The product is determined by a host/cookie condition, but[ otherwise they get grouped under `*`](https://github.com/hashicorp/dev-portal/blob/main/build-libs/redirects.js#L420-L432), which is the [fallback value](https://github.com/hashicorp/dev-portal/blob/main/src/middleware.ts#L24) when running the middleware.

Reply:
Updated it!